packages/server/parseFormBody.ts

```diff
@@ -0,0 +1,80 @@
+import {HttpResponse, HttpRequest} from 'uWebSockets.js'
+import Busboy from 'busboy'
+import {Readable} from 'stream'
+import {OutgoingMessage} from '@mattkrick/graphql-trebuchet-client'
+import {APP_MAX_AVATAR_FILE_SIZE} from 'parabol-client/utils/constants'
+
+const bodyStream = (res: HttpResponse) => {
+  const stream = new Readable()
+  stream._read = () => 'no-op'
+  res.onData((ab, isLast) => {
+    const curBuf = Buffer.from(ab)
+    stream.push(new Uint8Array(curBuf.slice(curBuf.byteOffset, curBuf.byteLength)))
+    if (isLast) stream.push(null)
+  })
+  return stream
+}
+
+const reqHeaders = (req: HttpRequest) => {
+  const headers = {} as {[key: string]: string}
+  req.forEach((k, v) => (headers[k] = v))
+  return headers
+}
+
+const parseFile = (fileStream, contentType) => {
+  return new Promise((resolve, reject) => {
+    let buffer: Buffer
+    fileStream.on('data', (curBuf) => {
+      buffer = buffer ? Buffer.concat([buffer, curBuf]) : Buffer.concat([curBuf])
+    })
+    fileStream.on('end', () => {
+      const parsedFile = {
+        buffer,
+        contentType
+      } as {
+        contentType: string
+        buffer: Buffer
+      }
+      resolve(parsedFile)
```

Comment:
parses the entire file before sending to => redis => graphql executor. i think we could technically use [Redis Streams](https://redis.io/topics/streams-intro) for the optimal solution. this was simple and works, so i'll see what tensions come up.

Reply:
i'm so glad you brought this up! i've been thinking about this too--- do we send the whole thing to graphql? do we stream it to redis? do we store it to disk here & then just return a file path to graphql? @adaniels-parabol probably has better thoughts than me on this.

Reply:
> do we store it to disk here & then just return a file path to graphql

^^ i think that would work for the on-premise local file system scenario, but not for Parabol Cloud S3 scenario. The reason being because the main server and gqlexecutor may not always be on the same machine.

i think part of the difficulty is we're really trying to solve the problem with a unified solution for different deployment scenarios.

> do we send the whole thing to graphql?

^^ do you mean directly to gqlexecutor? how would this work?

the one thing i like about sending it all to redis is that we don't make any assumptions about what we want to do with the contents until we hit the graphql resolver.